### DISCUSSION TITLE
Nuclear Hazard Theme

### DISCUSSION THREAD

**Discussion by StigtriX** [nukeop/nuclear#1046] [Show and tell]
I made a theme with an alternative colour scheme which is more "nuclear" so to speak. I have not found a proper use for the yellow colour yet, apart from a gradient in the logo. The letters and icons on buttons are black instead of white for this theme.

![Nuclear - Hazard Green - UI](https://user-images.githubusercontent.com/58557127/131396912-9fa79172-d1d3-4183-9797-ef7c8bf961a5.png)

Will you support themes in the future? 🙂

**Comment by nukeop** [comment]
Thanks, this looks pretty cool, although the yellow/white combination in the logo is too low contrast imo, the second logo should be used in this case. Maybe at some point we will change the logo in the program to one only showing the triangles, so it fits any background.

We don't have a way to support theming yet, but it might be coming in the future. It would require some pretty significant effort to add it though.

If you're feeling like it, you can try building your own version by modifying the palette here: 
https://github.com/nukeop/nuclear/blob/6a99b3f85042f170de840d3e9b9381d8cd5fd84b/packages/app/app/vars.scss
https://github.com/nukeop/nuclear/blob/6a99b3f85042f170de840d3e9b9381d8cd5fd84b/packages/ui/lib/common.scss

There might be colors hardcoded in some places, but most components use colors from these palettes. I think we should replace this with a single scss file that defines the colors in the future, so we don't need to keep the duplicate in sync.

**Comment by StigtriX** [comment]
Thank you for the kind words and the feedback. I made a new logo with darker shadows and also included versions with triangles only.
![Nuclear - Hazard Green - UI - 1 1](https://user-images.githubusercontent.com/58557127/132003342-1620b6fb-47ba-4abd-bafb-550e68a37a7b.png)

I dabble in design and not coding, but I can have a look and see if there is anything I can do :)

**Comment by StigtriX** [comment]
I don't think I am capable of building my own version yet 😅 

But I thought of something relevant: why not add a default icon for songs when there is no album art present? The colours can match the selected theme.

![Nuclear - No Cover Default Picture](https://user-images.githubusercontent.com/58557127/133130707-99967168-5b5c-40b6-b34c-e7cfc7239d53.png)

**Comment by nukeop** [comment]
There already is a default icon though; the star picture comes from last.fm - I haven't found a good source of images for those individual tracks yet (they disabled pictures on some of their endpoints some time ago).

**Comment by shablin** [comment]
Is themes still not supported?

**Reply by nukeop** [reply]
Hello, theming will be supported in the next iteration of Nuclear, which is currently being developed here: https://github.com/NuclearPlayer/nuclear-xrd

When it's ready, I will replace this repo's contents with this new version.
